### ISSUE TITLE
[MJAVADOC-783] Invalid path when using TagletArtifact and TagletPath

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#1161]
**[Rob Gordon](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER303448)** opened **[MJAVADOC-783](https://issues.apache.org/jira/browse/MJAVADOC-783?redirect=false)** and commented

I’m trying to use the Javadoc plugin with both a tagletartifact and a tagletpath but I get an InvliadPathException from the Javadoc goal.

I can see from the generated options file that there is no separator between the end of the artifact path and the taglet path.

I think the problem is that getTagletPath method of AbstractJavadocMojo should check for an existing path before adding the tagletpath. Something like:

```java
        if (tagletpath != null && !tagletpath.isEmpty()) {
            if (path.length() > 0) {
                 path.append(File.pathSeparator);
           }

            path.append(JavadocUtil.unifyPathSeparator(tagletpath));
        }
```

I looked at creating a pull request but I can’t work out how to create a suitable test. I found a taglet-test-plugin-config.xml, but not where it’s used.

Could someone familiar with this codebase possibly take a look at this problem please?

My configuration can be seen here: https://github.com/robjg/oj-parent/blob/d3e2328ca285516f1a4f6ee2bce94b44ba89b748/pom.xml#L304

I attempted to report this problem via 'users@maven.apache.org' however my mail did not make it. 

---

**Affects:** 3.6.3

**Issue Links:**
- [MJAVADOC-775](https://issues.apache.org/jira/browse/MJAVADOC-775) Option 'taglets/taglet/tagletpath' ignored when pointing to a JAR
 (_**"is duplicated by"**_)

**Remote Links:**
- [GitHub Pull Request #255
](https://github.com/apache/maven-javadoc-plugin/pull/255)
- [GitHub Pull Request #300
](https://github.com/apache/maven-javadoc-plugin/pull/300)

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Can you provide a sample project which cases the exception?

**Comment by jira-importer** [comment]
**[Alexander Kriegisch](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kriegaex)** commented

> I attempted to report this problem via 'users@maven.apache.org' however my mail did not make it.

Even after a quarter of the 21st century has passed, Apache mailing lists stubbornly insist on only accepting plain text messages. I am sure, that saves the Internet bandwidth-wise and the planet as a whole, and everyone still reads e-mails on text consoles. I actually also like to write plain text messages, even in business. But sometimes, it makes sense to have some formatting for inline code blocks and them not being butchered by word wrap, or to insert some explanatory inline pics, making the message so much easier and quicker to read than attaching same pics and code blocks as extra files which the reader then has to open one by one, trying to fully understand the message.

An organisation at the forefront of OSS development, fostering an innovative spirit, should think about whether it makes sense to hang on to anachronisms like that at the danger of becoming an anachronism itself one day.

Sorry to all sensitive souls for the snarky comment, but I could not resist and am gladly taking all your blame.

**Comment by jira-importer** [comment]
**[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo)** commented

Call me a luddite, but developers work in plain text for good reason, and that should include email. We'd all be better off if HTML email had never been invented.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I fully agree with [Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo), HTML is the worst ever happened to email. We all know who's fault is this.

**Comment by jira-importer** [comment]
**[Alexander Kriegisch](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kriegaex)** commented

[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo), no I won't call you a luddite.

[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo) and [Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o), I do not wish to start a general debate about the pros and cons of HTML mail here, so please forgive me for my digression above. Anyway, I cannot help but notice that, while the two of you agree, I gave reasons for my statement, but the two of you simply stated something as if it was a fact, which it is not. It is just an opinion, and both of you expressed your respective opinions as what in German we call "killer phrases".

_"Even when the experts all agree, they may well be mistaken." -- Bertrand Russell_

**Comment by jira-importer** [comment]
**[Alexander Kriegisch](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kriegaex)** commented

Fixed by https://github.com/apache/maven-javadoc-plugin/pull/255. I gave [robjg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=robjg) proper credit as a co-author of the commit fixing the same problem as his suggestion, just in a syntactically slightly different way.

**Comment by jira-importer** [comment]
**[Rob Gordon](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER303448)** commented

Thanks Alexander, I built the plugin from your clone - it works fine. Many thanks for fixing this.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [be2fa208674c781b08658878ae375f2c058992fc](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=be2fa208674c781b08658878ae375f2c058992fc).
